datafusion/expr/src/function.rs

```diff
@@ -185,14 +188,14 @@ pub fn return_type(
                 "The {fun} function can only accept list as the first argument"
             ))),
         },
-        BuiltinScalarFunction::Cardinality => Ok(DataType::UInt64),
-        BuiltinScalarFunction::MakeArray => Ok(DataType::List(Arc::new(Field::new(
+        BuiltinScalarFunction::Cardinality => Ok(UInt64),
```

Comment:
This is a nice change, but will conflict with https://github.com/apache/arrow-datafusion/pull/6612

(which is fine, I can handle resolving it, but perhaps you have some time to review https://github.com/apache/arrow-datafusion/pull/6612 @jackwener  to minimize future conflicts)?